source/Outfit.cpp

```diff
@@ -244,6 +263,29 @@ void Outfit::Load(const DataNode &node)
 
 
 
+void Outfit::FinishLoading()
+{
+	const Outfit *parent = GameData::Outfits().Get(parentName);
+		
+	// Copy data over from the parent outfit.
+	if(!hasCategory)
+		category = parent->Category();
+	if(!hasThumbnail)
+		thumbnail = parent->Thumbnail();
+	if(!hasCost)
+		cost = parent->Cost();
+	if(!hasMass)
+		mass = parent->Mass();
+	
+	for(const auto &it : parent->Attributes())
+	{
+		if(!attributes.Has(it.first))
```

Comment:
You shouldn't be adding new functions to the dictionary class. Dictionary::Get can be used exactly like how you're using this Dictionary::Has function that you created.
```suggestion
		if(!attributes.Get(it.first))
```
And looking to ship once more, ship only overwrites any attribute if the variant has no attributes. Something similar should be done here where only if attributes is empty do the parent attributes get used (and in that case you can simply assign the parent attribute object to the variant attribute object instead of iterating through all the attributes).
https://github.com/endless-sky/endless-sky/blob/c9a79a3080fc7a52d1123385f5fdafc451fc73f3/source/Ship.cpp#L496-L497

Reply:
The intent here is likely to facilitate "intentional zero" values being supported.

Reply:
That's correct.